File: llm_generated_code/ChatGPT/Python/Level2/Q8.py
def max_meetings(start, end):
    # Pair each meeting's start and end times, along with its original index
    meetings = sorted(zip(start, end, range(len(start))), key=lambda x: x[1])

    max_meetings_order = []
    last_end_time = 0

    for s, e, idx in meetings:
        if s >= last_end_time:  # If the meeting starts after or when the last meeting ended
            max_meetings_order.append(idx)  # Add meeting index to the order
            last_end_time = e  # Update the end time to this meeting's end time

    return len(max_meetings_order), [start[i] for i in max_meetings_order], [end[i] for i in max_meetings_order]

# Example usage
start_times = [1, 3, 0, 5, 8, 5]
end_times = [2, 4, 6, 7, 9, 9]

max_meeting_count, ordered_start_times, ordered_end_times = max_meetings(start_times, end_times)
print("Maximum number of meetings:", max_meeting_count)
print("Order of meetings (start times):", ordered_start_times)
print("Order of meetings (end times):", ordered_end_times)

File: llm_generated_code/ChatGPT/CPP/Level2/Q8.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

struct Meeting {
    int start, end, index;
};

bool compare(Meeting m1, Meeting m2) {
    return m1.end < m2.end;  // Sort by end time
}

void maximizeMeetings(vector<int>& start, vector<int>& end) {
    int n = start.size();
    vector<Meeting> meetings;

    // Store meeting details with index
    for (int i = 0; i < n; i++) {
        meetings.push_back({start[i], end[i], i + 1});
    }

    // Sort meetings based on end times
    sort(meetings.begin(), meetings.end(), compare);

    // Select meetings
    vector<int> selectedMeetings;
    int lastEnd = 0;

    for (const auto& meeting : meetings) {
        if (meeting.start >= lastEnd) {  // If the meeting doesn't overlap
            selectedMeetings.push_back(meeting.index);
            lastEnd = meeting.end;  // Update end time for the last selected meeting
        }
    }

    // Output results
    cout << "Maximum number of meetings: " << selectedMeetings.size() << endl;
    cout << "Order of meetings to maximize usage: ";
    for (int i : selectedMeetings) {
        cout << i << " ";
    }
    cout << endl;
}

int main() {
    vector<int> start = {1, 3, 0, 5, 8, 5};
    vector<int> end = {2, 4, 6, 7, 9, 9};

    maximizeMeetings(start, end);

    return 0;
}

